spyderlib/widgets/editor.py

```diff
@@ -64,106 +161,175 @@ def __init__(self, parent, tabs, fullpath_sorting):
+        
         edit_layout = QHBoxLayout()
-        edit_layout.addWidget(flabel)
         edit_layout.addWidget(self.edit)
-        edit_layout.addWidget(fhint)
-
+        
+        self.edit_line.connect(self.handle_edit_line)
         self.listwidget = QListWidget(self)
-        self.listwidget.setResizeMode(QListWidget.Adjust)
+        self.listwidget.setItemDelegate(HTMLDelegate(self))
         self.listwidget.itemSelectionChanged.connect(self.item_selection_changed)
-        self.listwidget.itemActivated.connect(self.edit_file)
-
-        btn_layout = QHBoxLayout()
-        edit_btn = create_toolbutton(self, icon=get_icon('edit.png'),
-                     text=_("&Edit file"), autoraise=False,
-                     triggered=self.edit_file, text_beside_icon=True)
-        edit_btn.setMinimumHeight(28)
-        btn_layout.addWidget(edit_btn)
-
-        btn_layout.addStretch()
-        btn_layout.addSpacing(150)
-        btn_layout.addStretch()
-
-        close_btn = create_toolbutton(self, text=_("&Close file"),
-              icon=get_icon("fileclose.png"),
-              autoraise=False, text_beside_icon=True,
-              triggered=lambda: self.close_file.emit(
-                                  self.indexes[self.listwidget.currentRow()]))
-        close_btn.setMinimumHeight(28)
-        btn_layout.addWidget(close_btn)
-
-        hint = QLabel(_("Hint: press <b>Alt</b> to show accelerators"))
-        hint.setAlignment(Qt.AlignCenter)
+        self.listwidget.itemActivated.connect(self.handle_edit_file)
+        self.original_path = None
+        self.original_line_num = None
+        self.line_num = -1
+        self.line_num_modified_for_path = None
+        
+        self.rejected.connect(self.handle_rejection)
 
+        hint_text = QLabel(_("Press <b>Enter</b> to switch files or <b>Esc</b> to cancel.<br>" +\
+                            "Type to filter filenames. Use <b>:number</b> to go to a line, e.g. 'main:42'.<br>" +\
+                            "Press <b>Ctrl+W</b> to close current tab."))
+        hint_text.setAlignment(Qt.AlignCenter)
         vlayout = QVBoxLayout()
         vlayout.addLayout(edit_layout)
         vlayout.addWidget(self.listwidget)
-        vlayout.addLayout(btn_layout)
-        vlayout.addWidget(hint)
+        vlayout.addWidget(hint_text)
         self.setLayout(vlayout)
 
-        self.tabs = tabs
-        self.fullpath_sorting = fullpath_sorting
-        self.buttons = (edit_btn, close_btn)
+        self.edit.selectAll()
+        self.edit.setFocus() 
+        geo = parent.geometry()
+        width = min(300,0.8*geo.width())
+        self.listwidget.setMinimumWidth(width)
+
+        left = parent.geometry().width()/2 - width/2
+        top = 0
+        while parent:
+            geo = parent.geometry()
+            top += geo.top()
+            left += geo.left()
+            parent = parent.parent()
+        self.move(left, top + self.tabs.tabBar().geometry().height() + 1) #the +1 is a hack

```

Comment:
It may depend on the theme used. That said I don't know how to compute a better value.